src/Service/UserAvatarManager.php

```diff
@@ -0,0 +1,122 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\Image;
+use App\Entity\User;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpFoundation\File\UploadedFile;
+
+class UserAvatarManager
+{
+    /**
+     * @var ImageCropperResizer
+     */
+    private $cropperResizer;
+    /**
+     * @var string
+     */
+    private $targetDirectory;
+
+    /**
+     * @var int
+     */
+    private $profilePicWidth;
+
+    /**
+     * @var int
+     */
+    private $profilePicHeight;
+
+    /**
+     * @var int
+     */
+    private $thumbnailPicWidth;
+
+    /**
+     * @var int
+     */
+    private $thumbnailPicHeight;
+
+    public function __construct(
+        string $targetDirectory,
+        array $avatarSize,
+        ImageCropperResizer $cropperResizer
+    ) {
+        $this->cropperResizer = $cropperResizer;
+        $this->targetDirectory = $targetDirectory;
+        $this->profilePicWidth = $avatarSize['profile']['width'];
```

Comment:
You should loop over the `$avatarSize` so that it's not necessary to modify the code if you need a new size.